src/ucp/wireup/select.c

```diff
@@ -777,7 +795,17 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_add_lane_desc(
         UCS_BITMAP_SET(select_ctx->tl_bitmap, select_info->rsc_index);
     }
 
-out_update_score:
+out_update:
+    if (lane_desc->path_index == UCP_WIREUP_PATH_INDEX_UNDEFINED) {
+        lane_desc->path_index = select_info->path_index;
+    } else {
+        ucs_assertv_always(
```

Comment:
this assert seems redundant since we get here only if pathIndex was not set or we passed the ucp_wireup_check_path_index check